UNKNOWN_ERROR after updating to TestNG 7.0.0

After upgrading to TestNG 7.0.0, my mutation test (using PIT 1.4.9) fails with the following error:
```

8:08:11 PM PIT >> SEVERE : Coverage generator Minion exited abnormally due to UNKNOWN_ERROR
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  23.208 s
[INFO] Finished at: 2019-08-19T20:08:11Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.pitest:pitest-maven:1.4.9:mutationCoverage (default-cli) on project BetterRandom-Java7: Execution default-cli of goal org.pitest:pitest-maven:1.4.9:mutationCoverage failed: Coverage generation minion exited abnormally!
[ERROR] 
[ERROR] Please copy and paste the information and the complete stacktrace below when reporting an issue
[ERROR] VM : OpenJDK 64-Bit Server VM
[ERROR] Vendor : Azul Systems, Inc.
[ERROR] Version : 25.222-b10
[ERROR] Uptime : 28225
[ERROR] Input -> 
[ERROR]  1 : -Dclassworlds.conf=/usr/share/apache-maven-3.6.1/bin/m2.conf
[ERROR]  2 : -Dmaven.home=/usr/share/apache-maven-3.6.1
[ERROR]  3 : -Dlibrary.jansi.path=/usr/share/apache-maven-3.6.1/lib/jansi-native
[ERROR]  4 : -Dmaven.multiModuleProjectDirectory=/home/vsts/work/1/s/betterrandom
[ERROR] BootClassPathSupported : true
[ERROR] 
[ERROR] 
[ERROR] Please copy and paste the information and the complete stacktrace below when reporting an issue
[ERROR] VM : OpenJDK 64-Bit Server VM
[ERROR] Vendor : Azul Systems, Inc.
[ERROR] Version : 25.222-b10
[ERROR] Uptime : 28229
[ERROR] Input -> 
[ERROR]  1 : -Dclassworlds.conf=/usr/share/apache-maven-3.6.1/bin/m2.conf
[ERROR]  2 : -Dmaven.home=/usr/share/apache-maven-3.6.1
[ERROR]  3 : -Dlibrary.jansi.path=/usr/share/apache-maven-3.6.1/lib/jansi-native
[ERROR]  4 : -Dmaven.multiModuleProjectDirectory=/home/vsts/work/1/s/betterrandom
[ERROR] BootClassPathSupported : true
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```
Worked fine with TestNG 6.14.3.

https://dev.azure.com/polymorpheus/BetterRandom/_build/results?buildId=955

--- a/pitest-maven-verification/src/test/resources/pit-testng-jmockit/pom.xml
+++ b/pitest-maven-verification/src/test/resources/pit-testng-jmockit/pom.xml
@@ -16,7 +16,7 @@
         <dependency>
             <groupId>org.testng</groupId>
             <artifactId>testng</artifactId>
-            <version>6.8.8</version>
+			<version>6.9.10</version>
             <scope>test</scope>
         </dependency>
         <dependency>

--- a/pitest-maven-verification/src/test/resources/pit-testng/pom.xml
+++ b/pitest-maven-verification/src/test/resources/pit-testng/pom.xml
@@ -10,7 +10,7 @@
 		<dependency>
 			<groupId>org.testng</groupId>
 			<artifactId>testng</artifactId>
-			<version>6.1.1</version>
+			<version>6.9.10</version>
 			<scope>test</scope>
 		</dependency>
 	</dependencies>

--- a/pitest/src/main/java/org/pitest/testng/TestNGTestUnit.java
+++ b/pitest/src/main/java/org/pitest/testng/TestNGTestUnit.java
@@ -26,6 +26,7 @@ import org.testng.IInvokedMethod;
 import org.testng.IInvokedMethodListener;
 import org.testng.ITestContext;
 import org.testng.ITestListener;
+import org.testng.ITestNGListener;
 import org.testng.ITestResult;
 import org.testng.SkipException;
 import org.testng.TestNG;
@@ -47,8 +48,9 @@ public class TestNGTestUnit extends AbstractTestUnit {
   private static final MutableTestListenerWrapper LISTENER = new MutableTestListenerWrapper();
 
   static {
-    TESTNG.addListener(LISTENER);
-    TESTNG.addInvokedMethodListener(new FailFast(LISTENER));
+    // force using TestNG.addListener(ITestNGListener) to be compatible with TestNG 7.x
+    TESTNG.addListener((ITestNGListener)LISTENER);
+    TESTNG.addListener((ITestNGListener)new FailFast(LISTENER));
   }
 
   private final Class<?>                     clazz;

--- a/pom.xml
+++ b/pom.xml
@@ -297,7 +297,7 @@
 		<maven.version>2.2.1</maven.version>
 		<powermock.version>1.7.3</powermock.version>
 		<surefire.version>2.17</surefire.version>
-		<testng.version>6.8.21</testng.version>
+		<testng.version>6.9.10</testng.version>
 		<slf4j.version>1.7.12</slf4j.version>
 
 		<maven.failsafe-plugin.version>${surefire.version}</maven.failsafe-plugin.version>

